Add IEx section to debugging lesson

brain-geek opened the pull request:
Hey guys, could you review this?

Closes #1643 

@elixirschool/english

SophieDeBenedetto commented on en/lessons/specifics/debugging.md:
````diff
@@ -8,6 +8,113 @@ In this lesson we'll learn about debugging Elixir code as well as static analysi
+34
+```
+
+But now let's get to the exciting part - the debugging. Put `require IEx; IEx.pry` in the line after `b = 0` and let's try running it once again. You'll get something like this:
````
Let's add a snippet here that shows the `sum/2` function with the `require IEx; IEx.pry` on the appropriate line.

brain-geek replied:
I don't think that's necessary. We would duplicate the same listing with one line changed. Why do you think this is unclear?

SophieDeBenedetto replied:
Actually on second thought, you're right. I think it is clear enough as-is. Thanks!

SophieDeBenedetto commented on en/lessons/specifics/debugging.md:
````diff
@@ -8,6 +8,113 @@ In this lesson we'll learn about debugging Elixir code as well as static analysi
+34
+```
+
+You should note that vital message. When running an application, as usual, IEx outputs the error message instead of blocking the application. To run it properly you need to prepend your command with `iex -S`. For example, `iex -S mix phx.server` to debug your Phoenix application. In our case, it's going to be `iex -r test.exs` to require the file:
````
I wouldn't call this an error message, maybe let's say `IEx outputs this message instead of blocking execution of the program`.

SophieDeBenedetto replied:
Could you add a short explanation of `iex -S`? Why does this fix the problem/allows us to use `IEx.pry`?

SophieDeBenedetto replied:
`In our case, it's going to be `iex -r test.exs` to require the file:` should be `In our case, it's going to be `iex -S test.exs` to require the file:`

brain-geek replied:
I believe all three points have been addressed.

SophieDeBenedetto commented on en/lessons/specifics/debugging.md:
````diff
@@ -8,6 +8,113 @@ In this lesson we'll learn about debugging Elixir code as well as static analysi
+After responding to the prompt via `y` or pressing Enter, you've entered the interactive mode.
+
+```
+ $ iex -r test.exs
````
`iex -S test.exs`

brain-geek replied:
```
 $ iex -S test.exs
Erlang/OTP 21 [erts-10.3.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]

-S : Could not find executable test.exs
 $ iex -r test.exs

Erlang/OTP 21 [erts-10.3.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace]

warning: variable "b" is unused (if the variable is not meant to be used, prefix it with an underscore)
  test.exs:2

Request to pry #PID<0.107.0> at TestMod.sum/1 (test.exs:5)

    3:     b = 0
    4:
    5:     require IEx; IEx.pry
    6:
    7:     a + b

Allow? [Yn]
```

SophieDeBenedetto replied:
Ah! interesting. I've never seen the `-r` flag before--what does it do?

brain-geek replied:
It requires/loads the specified file.

SophieDeBenedetto commented:
Hey @brain-geek, just a ping on this. So excited to have this section published--I think its something we've needed for a long time 🚀 

If you don't have time to address the minor comment I left above, I'm inclined to approve and merge and we can always revisit this later if we want to. Let me know!

brain-geek commented:
@SophieDeBenedetto Thanks for the feedback! I've addressed all the points.

Yeah, I rarely have time to work on Elixir school recently, so if there would be some minor changes - feel free to add the changes and merge it yourself. I've added you as collaborator to my repository fork for that.
